node/modules/ipfsclient.go

```diff
@@ -6,21 +6,38 @@ import (
 
 	"github.com/ipfs/go-filestore"
 	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/multiformats/go-multiaddr"
 
 	"github.com/filecoin-project/lotus/lib/bufbstore"
 	"github.com/filecoin-project/lotus/lib/ipfsbstore"
 	"github.com/filecoin-project/lotus/node/modules/dtypes"
 	"github.com/filecoin-project/lotus/node/modules/helpers"
 )
 
-func IpfsClientBlockstore(mctx helpers.MetricsCtx, lc fx.Lifecycle, fstore dtypes.ClientFilestore) (dtypes.ClientBlockstore, error) {
-	ipfsbs, err := ipfsbstore.NewIpfsBstore(helpers.LifecycleCtx(mctx, lc))
-	if err != nil {
-		return nil, xerrors.Errorf("constructing ipfs blockstore: %w", err)
+// IpfsClientBlockstore returns a ClientBlockstore implementation backed by an IPFS node.
+// If ipfsMaddr is empty, a local IPFS node is assumed considering IPFS_PATH configuration.
+// If ipfsMaddr is not empty, it will connect to the remote IPFS node with the provided multiaddress.
+// The flag useForRetrieval indicates if the IPFS node will also be used for storing retrieving deals.
+func IpfsClientBlockstore(ipfsMaddr string, useForRetrieval bool) func(helpers.MetricsCtx, fx.Lifecycle, dtypes.ClientFilestore) (dtypes.ClientBlockstore, error) {
+	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, fstore dtypes.ClientFilestore) (dtypes.ClientBlockstore, error) {
+		ma, err := multiaddr.NewMultiaddr(ipfsMaddr)
+		if err != nil {
+			return nil, xerrors.Errorf("parsing ipfs multiaddr: %w", err)
+		}
+		var ipfsbs *ipfsbstore.IpfsBstore
+		if ipfsMaddr != "" {
```

Comment:
I'd expect this to be 
```suggestion
		if ipfsMaddr != "" {
			ma, err := multiaddr.NewMultiaddr(ipfsMaddr)
			if err != nil {
				return nil, xerrors.Errorf("parsing ipfs multiaddr: %w", err)
			}
			var ipfsbs *ipfsbstore.IpfsBstore
```

Reply:
Oh, yeah. Will change.